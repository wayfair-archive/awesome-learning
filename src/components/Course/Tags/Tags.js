import React from 'react';
import PropTypes from 'prop-types';
import StyledLink from '../../shared/StyledLink';
import './Tags.scss';

const Tags = ({ tags, tagSlugs }) => (
  <div className="Tags">
    <ul className="Tags-list">
      {tagSlugs.map((slug, i) => (
        <li className="Tags-listItem" key={`${tags[i]}slug`}>
          <StyledLink variation="pill" path={slug}>
            {tags[i]}
          </StyledLink>
        </li>
      ))}
    </ul>
  </div>
);

Tags.propTypes = {
  tags: PropTypes.array.isRequired,
  tagSlugs: PropTypes.array.isRequired
};

export default Tags;
