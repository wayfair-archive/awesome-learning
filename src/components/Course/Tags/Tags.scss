@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.Tags {
  @include margin-bottom(.5);
  &-list {
    list-style: none;
    margin: 0 -10px;
    padding: 0;
  }
  &-listItem {
    display: inline-block;
    margin: 10px 5px;
  }
}