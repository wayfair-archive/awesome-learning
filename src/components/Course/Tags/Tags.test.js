import React from 'react';
import renderer from 'react-test-renderer';
import Tags from './Tags';

describe('Tags', () => {
  it('renders correctly', () => {
    const props = {
      tags: [
        'test_0',
        'test_1'
      ],
      tagSlugs: [
        '/test_0',
        '/test_1'
      ]
    };

    const tree = renderer.create(<Tags {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
