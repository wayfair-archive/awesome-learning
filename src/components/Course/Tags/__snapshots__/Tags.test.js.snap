// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
<div
  className="Tags"
>
  <ul
    className="Tags-list"
  >
    <li
      className="Tags-listItem"
    >
      <Link
        activeClassName=""
        className="Link Link--pill"
        to="/test_0"
      >
        test_0
      </Link>
    </li>
    <li
      className="Tags-listItem"
    >
      <Link
        activeClassName=""
        className="Link Link--pill"
        to="/test_1"
      >
        test_1
      </Link>
    </li>
  </ul>
</div>
`;
